cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,38 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
+      0, TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()),
+      Integer.MAX_VALUE, "system.collect.bytes",
+      MetricType.COUNTER,
+      ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, streamId.getNamespaceId(),
+                      Constants.Metrics.Tag.STREAM, streamId.getName()),
+      ImmutableList.<String>of()
+    );
+
+    try {
+      Collection<MetricTimeSeries> metrics = metricStore.query(metricDataQuery);
+      if (metrics == null || metrics.size() != 1) {
```

Comment:
if the data is not available yet, you would get an empty time series list, might be better to return 0 in that case ? what do you think ?